### PR TITLE
feat(launchpad): initial automatic resource allocation logic

### PR DESCRIPTION
**Pull request by RolandSherwin**
## Description

reviewpad:summary

### CODE REVIEW

File: node-launchpad/.config/config.json5 [hunk 1]
```diff
@@ -1,27 +1,39 @@
...
     "Home": {
-      "<a>": {"HomeActions":"AddNode"}, // Add nodes
-      "<s>": {"HomeActions":"StartNodes"}, // Start nodes
-      "<k>": {"HomeActions":"StopNode"}, // Stop nodes
-      "<r>": {"HomeActions":"RemoveNode"},
-      "<d>": {"HomeActions":"InputDiscordUsername"},
+      "<Ctrl-g>": {"HomeActions":"StartNodes"},
+      "<Ctrl-G>": {"HomeActions":"StartNodes"},
+      "<Ctrl-x>": {"HomeActions":"StopNodes"}, 
+      "<Ctrl-X>": {"HomeActions":"StopNodes"},
```
Comment (mazzi) [inline node-launchpad/.config/config.json5]: Little linting. Blank in EOL.

File: node-launchpad/src/components/home.rs [hunk 2]
```diff
@@ -328,72 +377,106 @@ impl Component for Home {
...
+fn add_and_start_nodes(
+    count: usize,
+    mut nodes_to_start: Vec<String>,
+    action_sender: UnboundedSender<Action>,
+) {
+    let peers = PeersArgs::default(); // will fetch from network contacts
+
+    tokio::task::spawn_local(async move {
+        let result = sn_node_manager::cmd::node::add(
+            Some(count as u16),
```
Comment (joshuef) [inline node-launchpad/src/components/home.rs:437]: should this not be `nodes_to_start` ?

**Review by joshuef — CHANGES_REQUESTED** [review]
I think we need the `add_multiple_based_upon_storage` concept in the nodemanager itself?

So the TUI would just be calling that with provided value? 

(And the nodeman can just have it passed as a flag to get the same result?

File: node-launchpad/src/components/home.rs (hunk 2 above)
Comment (RolandSherwin) [reply node-launchpad/src/components/home.rs:437]: `nodes_to_start` denotes the inactive nodes that we are supposed to restart basically.

**Comment by joshuef** [comment]
Couple issues:

Need to `clear` input when being reused.
- Start-nodes + allocate `10`
- Press `D` for discord.
- See: 

Screenshot 2024-05-13 at 10 26 07

------

`Stop node` stops all nodes. (May be desireable... just need to update text)


----

Why `ctrg+g` to start and not `ctrl+s`?

**Review by joshuef — APPROVED** [review]

**Pull request merged**